Diary/Controller/KeyboardViewController.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  KeyboardViewController.swift
+//  Diary
+//
+//  Created by 김동욱 on 2022/06/14.
+//
+
+import UIKit
+
+class KeyboardViewController: UIViewController {
+    var bottomContraint: NSLayoutConstraint?
+    
+    func setUpKeyboardNotification() {
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillAppear), name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    @objc private func keyboardWillAppear(_ notification: Notification) {
+        guard let keyboardBounds = notification.userInfo?["UIKeyboardBoundsUserInfoKey"] as? NSValue else { return }
+        
+        bottomContraint?.constant = -keyboardBounds.cgRectValue.height
+    }
+    
+    @objc private func keyboardWillHide() {
+        bottomContraint?.constant = .zero
+    }
+}
```

Comment:
`addObserver` 하는 부분은 있는데 `removeObserver`를 하는 부분은 없네요!!

되게 사소한 부분이지만 메모리 남을 것 같습니다!!! 언제 해제해주는 것이 좋을까요??

Reply:
https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver

> If your app targets iOS 9.0 and later or macOS 10.11 and later, you do not need to unregister an observer that you created with this function. If you forget or are unable to remove an observer, the system cleans up the next time it would have posted to it.

따로 제거하지 않아도 자동으로 없어지는 것으로 이해했습니다🤔